hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -377,11 +377,13 @@
       name: filesystem_support
       desc: '''
             Enable ECC and disable scrambling for all regions.
-            Initialize flash with random bits. Erase some random regions / pages and keep the pages.
-            Do write and read back test for random page and check
-             - If pages are in the erased page list, write data should be random and
+            Initialize flash with erased state (FlashMemInitSet).
+            Execute random number of writes without writing the same locatino twice.
+            Record all write locations(Minimum resolution of location is 8bytes).
+            After that, execute write and read back test for random page and check
+             - If the locations are in the written record, write data should be all 0 and
                readback data should match with the written data.
-             - If pages are not in the erase page, write data should be all 0 and
+             - If the locations are not in the written record, write data should be random and
```

Comment:
is there anything we can do to ensure ECC is enabled?

Reply:
Other than this?https://github.com/lowRISC/opentitan/blob/8f83ec912577c2967a68fcdf0bc944e69b6ccc79/hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_filesystem_support_vseq.sv#L27-L28

Reply:
yeah, because the main thing to make sure double write is possible with ECC enabled. 
Do you think there's any reason to do an ECC check (maybe 1-bit error) to make sure the its functioning before the second write to 0?

Reply:
> yeah, because the main thing to make sure double write is possible with ECC enabled. Do you think there's any reason to do an ECC check (maybe 1-bit error) to make sure the its functioning before the second write to 0?

We can do that. Any particular reason for 'before' the 2nd write? I think 'after' the 2nd write might be more interesting because,
compared to other 'short-cut' error injections, injecting error to flash device flips a bit through backdoor read/write.
So even after injection error, when the 2nd write to 0 comes, it will just overwrite what ever current value to '0'.

Reply:
ah sorry, i was using the before write injection mostly as a way to prove ECC is there. So i was thinking something like this sequence.

first write (random data) -> read verify first write -> inject 1 bit error -> read verify again (should still be correct) -> second write -> read verify all 0's -> inject one bit error again -> read verify again (should be all 0's correct). 

Does this seem reasonable to you? I'm not sure if I'm being overly paranoid or if this actually provides better coverage.

Reply:
Yes it make sense to me. Feel free to overly paranoid :)

Reply:
@tjaychen Test has been update as we discussed.